contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1604,25 +1652,58 @@ export const ContentNode = new TreeResource({
   getChannelId: getChannelFromChannelScope,
 
   /**
-   * Waits for copying of content nodes to complete for all ids referenced in `ids` array
-   * @param {string[]} ids
+   * Waits for copying of contentnode to complete for id referenced in `id`.
+   * @param {string} id
    * @returns {Promise<void>}
    */
-  waitForCopying(ids) {
-    const observable = Dexie.liveQuery(() => {
-      return this.table
+  waitForCopying(id, startingRev) {
+    const observable = Dexie.liveQuery(async () => {
+      let copy_success_flag = 0;
+      let change_error_flag = 0;
+
+      copy_success_flag = await this.table
         .where('id')
-        .anyOf(ids)
-        .filter(node => !node[COPYING_FLAG])
-        .toArray();
+        .equals(id)
+        .filter(node => node[COPYING_STATUS] === COPYING_STATUS_VALUES.SUCCESS)
+        .count();
+
+      if (copy_success_flag === 1) {
+        return {
+          copy_success_flag,
+          change_error_flag,
+        };
+      }
+
+      change_error_flag = await db[CHANGES_TABLE].where('[table+key]')
+        .equals([this.tableName, id])
+        .filter(change => {
+          if (
+            (change['errors'] || change['errors'] === '') &&
+            change['type'] === CHANGE_TYPES.COPIED
+          ) {
+            if (!startingRev || change['rev'] > startingRev) {
```

Comment:
what led to the addition of `startingRev` here?

Reply:
Sir, when we retry copying, there's a failed change object present in the indexed db for that node, so `startingRev` skips those old change objects.